src/python/gudhi/cubical_complex.pyx

```diff
@@ -58,16 +58,32 @@ cdef class CubicalComplex:
 
         Or
 
+        :param top_dimensional_cells: A multidimensional array of cells
+            filtration values.
+        :type top_dimensional_cells: numpy ndarray
+
+        Or
+
         :param perseus_file: A Perseus-style file name.
         :type perseus_file: string
         """
 
     # The real cython constructor
     def __cinit__(self, dimensions=None, top_dimensional_cells=None,
                   perseus_file=''):
-        if (dimensions is not None) and (top_dimensional_cells is not None) and (perseus_file == ''):
+        if ((dimensions is not None) and (top_dimensional_cells is not None)
+            and (perseus_file == '')):
             self.thisptr = new Bitmap_cubical_complex_base_interface(dimensions, top_dimensional_cells)
-        elif (dimensions is None) and (top_dimensional_cells is None) and (perseus_file != ''):
+        elif ((dimensions is None) and (top_dimensional_cells is not None)
+            and (perseus_file == '')):
+            if isinstance(top_dimensional_cells, np.ndarray):
```

Comment:
Why so strict? I think it would be convenient to accept lists in the example in cubical_complex_user.rst. `top_dimensional_cells=numpy.array(top_dimensional_cells,copy=false,order='F')` maybe?

Reply:
Yes you are right. I did the fix you propose.